File: app/src/main/java/com/abishov/hexocat/home/trending/TrendingModule.java
package com.abishov.hexocat.home.trending;

import com.abishov.hexocat.commons.dagger.FragmentScope;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
@FragmentScope
public abstract class TrendingModule {

    @Provides
    @FragmentScope
    static TrendingService trendingService(Retrofit retrofit) {
        return retrofit.create(TrendingService.class);
    }

    @Binds
    @FragmentScope
    abstract TrendingContract.Presenter trendingPresenter(TrendingPresenter impl);
}

File: app/src/main/java/com/abishov/hexocat/home/trending/TrendingModule.kt
package com.abishov.hexocat.home.trending

import com.abishov.hexocat.common.dagger.FragmentScope
import com.abishov.hexocat.common.dispatcher.DispatcherProvider
import com.apollographql.apollo.ApolloClient
import dagger.Module
import dagger.Provides

@Module
internal class TrendingModule {

  @Provides
  @FragmentScope
  internal fun trendingService(client: ApolloClient) = TrendingService(client)

  @Provides
  @FragmentScope
  internal fun trendingViewModelFactory(
    dispatcherProvider: DispatcherProvider,
    service: TrendingService
  ) = TrendingViewModelFactory(dispatcherProvider, service)
}
